docs/source/resources/graphql-glossary.md

```diff
@@ -0,0 +1,240 @@
+---
+title: GraphQL Glossary
+description: A comprehensive list of important GraphQL words and acronyms
+---
+
+
+We have put together a glossary of words and acronyms that you might hear or come across frequently in GraphQL land. Every developer and GraphQL enthusiast should be able to easily look up a particular term here when referenced in specs, reference guides, READMEs and within a codebase.
+
+
+<h2 id="Apollo">Apollo</h2>
+<p>A set of tools for building GraphQL apps. Some of these tools are Apollo Client (best way to use GraphQL to build Client apps), Apollo Server (for building GraphQL servers), Engine (performance monitoring and tracing), etc.</p>
+
+<h2 id="automatic-persisted-queries">Automatic Persisted Queries (APQ) </h2>
+<p> A technique for improving GraphQL network performance by reducing request size over the wire. </p>
+
+<h2 id="argument">Argument</h2>
+<p>A parameter or set of key-value pair passed to fields in a schema to filter out results sent back from the server to the client.</p>
+
+<h2 id="alias">Alias</h2>
+<p>An alternative name given to the result of a field to avoid conflicts during data fetching.</p>
+
+<h2 id="data-source">Data Source</h2>
+<p>A new utility and pattern for fetching and caching data from REST API endpoints. When layering GraphQL over REST APIs, data sources provides the ability to have a resource cache that saves and shares data easily across multiple GraphQL servers.</p>
+
+```js
+const { RESTDataSource } = require('apollo-datasource-rest');
+
+class MoviesAPI extends RESTDataSource {
+  constructor() {
+    super();
+    this.baseURL = 'https://movies-api.example.com/';
+  }
+
+  async getMovie(id) {
+    return this.get(`movies/${id}`);
+  }
+
+  async getMostViewedMovies(limit = 10) {
+    const data = await this.get('movies', {
+      per_page: limit,
+      order_by: 'most_viewed',
+    });
+    return data.results;
+  }
+}
+```
+
+<h2 id="deferred-query">Deferred Query</h2>
+<p>A declaration prefixed with an @ character that encapsulates programming logic for query execution on the client or server. There are built-in such as @skip, @include and custom directives. It can be used for features such as authentication, incremental data loading, etc.</p>
+
+<h2 id="docstring">Docstring</h2>
+<p>A technique for providing metadata to a GraphQL Document. It can used for describing types, fields and arguments.</p>
+
+```js
+"""
+Description for the User
+"""
+type User {
+  """
+  Description for first Name
+  """
+  firstName: String!
+
+  age(
+    """
+    Must be an integer
+    """
+    arg: Int
+  )
+}
+```
+
+<h2 id="document">Document</h2>
+<p>A file or request string that contains one or multiple definitions of a GraphQL type system and can be interpreted by a GraphQL execution engine.</p>
+
+<h2 id="extensions">Extensions </h2>
```

Comment:
Maybe we should leave this off? Extensions may not be supported/recommended after the request pipeline work @martijnwalraven is doing?